pkg/webhook/servicecatalog/servicebinding/validation/handle_reference_test.go

```diff
@@ -0,0 +1,203 @@
+			handler := validation.AdmissionHandler{}
+			handler.CreateValidators = []validation.Validator{&validation.ReferenceDeletion{}}
+			handler.UpdateValidators = []validation.Validator{&validation.ReferenceDeletion{}}
+
+			fakeClient := fake.NewFakeClientWithScheme(sch, &sc.ServiceInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      UpToDateInstance,
+					Namespace: namespace,
+				},
+			})
+
+			err := handler.InjectDecoder(decoder)
+			require.NoError(t, err)
+			err = handler.InjectClient(fakeClient)
+			require.NoError(t, err)
+
+			request.AdmissionRequest.Operation = test.operation
+
+			// When
+			response := handler.Handle(context.TODO(), request)
+
+			// Then
+			assert.True(t, response.AdmissionResponse.Allowed)
+		})
+	}
+}
+
+func TestAdmissionHandlerServiceInstanceReferenceOutOfDate(t *testing.T) {
+	//Given
+	namespace := "test-handler"
+	err := sc.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+
+	request := admission.Request{
+		AdmissionRequest: admissionv1beta1.AdmissionRequest{
+			UID:       "2222-bbbb",
+			Name:      "test-binding",
+			Namespace: namespace,
+			Kind: metav1.GroupVersionKind{
+				Kind:    "ServiceBinding",
+				Version: "v1beta1",
+				Group:   "servicecatalog.k8s.io",
+			},
+			Object: runtime.RawExtension{Raw: []byte(`{
+  				"apiVersion": "servicecatalog.k8s.io/v1beta1",
+  				"kind": "ServiceBinding",
+  				"metadata": {
+				  "finalizers": ["kubernetes-incubator/service-catalog"],
+  				  "creationTimestamp": null,
+  				  "name": "test-binding"
+  				},
+  				"spec": {
+				  "instanceRef": {
+					"name": "` + OutOfDateInstance + `"
+				  },
+				  "externalID": "123-abc",
+				  "secretName": "test-binding"
+  				}
+			}`)},
+		},
+	}
+
+	sch, err := sc.SchemeBuilderRuntime.Build()
+	require.NoError(t, err)
+
+	decoder, err := admission.NewDecoder(sch)
+	require.NoError(t, err)
+
+	tests := map[string]struct {
+		operation admissionv1beta1.Operation
+	}{
+		"Request for Create ServiceBinding should be denied": {
+			admissionv1beta1.Create,
+		},
+		"Request for Update ServiceBinding should be denied": {
+			admissionv1beta1.Update,
+		},
+	}
+
+	for desc, test := range tests {
+		t.Run(desc, func(t *testing.T) {
+			// Given
+			handler := validation.AdmissionHandler{}
+			handler.CreateValidators = []validation.Validator{&validation.ReferenceDeletion{}}
+			handler.UpdateValidators = []validation.Validator{&validation.ReferenceDeletion{}}
+
+			fakeClient := fake.NewFakeClientWithScheme(sch, &sc.ServiceInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              OutOfDateInstance,
+					Namespace:         namespace,
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+				},
+			})
+
+			err := handler.InjectDecoder(decoder)
+			require.NoError(t, err)
+			err = handler.InjectClient(fakeClient)
+			require.NoError(t, err)
+
+			request.AdmissionRequest.Operation = test.operation
+
+			// When
+			response := handler.Handle(context.TODO(), request)
```

Comment:
```suggestion
			response := handler.Handle(context.Background(), request)
```
`context.TODO()`  should be used only if you know that in future you will need to replace it with some more production impl. Here you will not do that.

same suggestion for all other occurrences in tests cases